src/screens/Lending/useCompoundSummaries.js

```diff
@@ -0,0 +1,41 @@
+// @flow
+import { useState, useEffect } from "react";
+import type { AccountLikeArray } from "@ledgerhq/live-common/lib/types";
+import type { CompoundAccountSummary } from "@ledgerhq/live-common/lib/compound/types";
+import { makeCompoundSummaryForAccount } from "@ledgerhq/live-common/lib/compound/logic";
+import { findCompoundToken } from "@ledgerhq/live-common/lib/currencies";
+
+import { isCompoundTokenSupported } from "@ledgerhq/live-common/lib/families/ethereum/modules/compound";
+
+const makeSummaries = (accounts: AccountLikeArray): CompoundAccountSummary[] =>
+  accounts
+    .map(acc => {
+      if (acc.type !== "TokenAccount") return null;
+      const ctoken = findCompoundToken(acc.token);
+      if (!ctoken) return null;
+
+      if (!isCompoundTokenSupported(ctoken)) return null;
+
+      const parentAccount = accounts.find(a => a.id === acc.parentId);
+      if (!parentAccount || parentAccount.type !== "Account") return null;
+      const summary = makeCompoundSummaryForAccount(acc, parentAccount);
+      return summary;
+    })
+    .filter(Boolean);
+
+export function useCompoundSummaries(
+  accounts: AccountLikeArray,
+): CompoundAccountSummary[] {
+  const [summaries, setSummaries] = useState(() => {
+    return makeSummaries(accounts);
+  });
+
+  useEffect(() => {
+    const newSummaries = makeSummaries(accounts);
+    if (newSummaries) {
+      setSummaries(newSummaries);
+    }
+  }, [accounts, setSummaries]);
+
+  return summaries;
+}
```

Comment:
can we unify these back into live-common and also consider merging with https://github.com/LedgerHQ/ledger-live-desktop/blob/develop/src/renderer/screens/lend/useCompoundSummaries.js ? no reason to duplicate these.

Reply:
https://github.com/LedgerHQ/ledger-live-common/pull/936 :+1:  i'll update the code once this is merged